File: backend/app/schemas/water.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WaterReservoirResponse(BaseModel):

    model_config = ConfigDict(
        from_attributes=True
    )

    id: int
    source_id: int | None

    district: str
    mandal: str
    reservoir: str
    river: str | None

    present_level_m: float | None
    present_level_ft: float | None

    present_capacity_mcum: float | None
    present_capacity_tmc: float | None

    frl_m: float | None
    frl_ft: float | None

    gross_capacity_mcum: float | None
    gross_capacity_tmc: float | None

    storage_percentage: float | None

    updated_at: datetime | None
    source: str
    fetched_at: datetime

File: frontend/src/pages/farmer/dashboard/types/water.ts
export interface Reservoir {
  id: number;
  source_id: number | null;

  district: string;
  mandal: string;
  reservoir: string;
  river: string | null;

  present_level_m: number | null;
  present_level_ft: number | null;

  present_capacity_mcum: number | null;
  present_capacity_tmc: number | null;

  frl_m: number | null;
  frl_ft: number | null;

  gross_capacity_mcum: number | null;
  gross_capacity_tmc: number | null;

  storage_percentage: number | null;

  updated_at: string | null;
  source: string;
  fetched_at: string;
}

export type WaterStatus =
  | "Good"
  | "Moderate"
  | "Low"
  | "Critical";
